src/index.ts

```diff
@@ -179,9 +225,16 @@ const AddKeyPlugin: Plugin = builder => {
       ...ObjectTypeDefinition(spec),
       ...Self.astNode,
     };
-    astNode.directives.push(
-      Directive("key", { fields: StringValue(nodeIdFieldName) })
-    );
+
+    const {
+      primaryKeyConstraint: { keyAttributes },
+    } = pgIntrospection;
+    const primaryKeyNames = keyAttributes.map(attr => inflection.column(attr));
+    const keyName = primaryKeyNames.length
+      ? primaryKeyNames.join(" ")
+      : nodeIdFieldName;
+
+    astNode.directives.push(Directive("key", { fields: StringValue(keyName) }));
```

Comment:
We should definitely support the `nodeId` by default; using the primary keys directly is not a best practice. Adding support for primary keys is desired if the node plugin is disabled, and it could also be used as an alternate fetcher. Fortunately according to the [Apollo federation spec](https://www.apollographql.com/docs/apollo-server/federation/federation-spec/#key) you can provide `@key` multiple times; so this is what we should do - once for nodeId, and once for the PKs.